password-hash/src/errors.rs

```diff
@@ -70,14 +83,27 @@ impl fmt::Display for Error {
             Self::OutputTooLong => f.write_str("PHF output too long (max 64-bytes)"),
             Self::ParamNameDuplicated => f.write_str("duplicate parameter"),
             Self::ParamNameInvalid => f.write_str("invalid parameter name"),
-            Self::ParamValueInvalid => f.write_str("invalid parameter value"),
+            Self::ParamValueInvalid(param_err) => match param_err {
+                ParamValueError::ToLong => f.write_str("invalid parameter value: value to long"),
+                ParamValueError::ToShort => f.write_str("invalid parameter value: value to short"),
+                ParamValueError::NotProvided => {
+                    f.write_str("invalid parameter value: required value not provided")
+                }
+                ParamValueError::InvalidChar => {
+                    f.write_str("invalid parameter value: contains invalid character")
+                }
+                ParamValueError::InvalidFormat => {
+                    f.write_str("invalid parameter value: value format is invalid")
+                }
+            },
```

Comment:
It seems like this could be more nicely factored as an `fmt::Display` impl on `ParamValueError` (or whatever it gets renamed to, if it's renamed)

Reply:
Oh yeah, that totally makes sense. Added.